core/src/test/java/org/infinispan/container/offheap/KeyGenerator.java

```diff
@@ -0,0 +1,78 @@
+package org.infinispan.container.offheap;
+
+import java.util.Random;
+
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.metadata.EmbeddedMetadata;
+import org.infinispan.metadata.Metadata;
+
+/**
+ * Generates keys and values to be used during the cache.
+ * These instances are pre-created to minimize impact
+ * during measurements.
+ * Strings are a popular choice, but trigger ad-hoc optimisations
+ * so we have options to test both with strings and other things.
```

Comment:
I'd like to see store as binary mentioned specifically wrt the ad-hoc optimizations (and any other places you know have optimizations for strings).

Reply:
Sorry this is a remnant of the JMH benchmark. Let me remove this jazz.